services/gundeck/src/Gundeck/Client.hs

```diff
@@ -14,17 +13,12 @@ import Data.Id
 import Data.Predicate
 import Gundeck.Monad
 import Gundeck.Push.Native
-import Gundeck.Util
-import Network.Wai (Request, Response)
+import Network.Wai (Response)
 import Network.Wai.Utilities
 
 import qualified Gundeck.Notification.Data as Notifications
 import qualified Gundeck.Push.Data         as Push
 
--- | REFACTOR: this is deprecated
-register :: UserId ::: ClientId ::: Request ::: JSON ::: JSON -> Gundeck Response
-register (_uid ::: _cid ::: _req ::: _) = return empty
```

Comment:
One thing to bear in mind here: removing this endpoint means that we _must_ release `brig` before `gundeck`, otherwise registering clients on `brig` would fail (using an endpoint that no longer returns `200`); for the sake of simplicity of deployment, I'd suggest leaving this endpoint in and removing it only after `brig` makes its way to production.